### DISCUSSION TITLE
Network initialisation

### DISCUSSION THREAD

**Discussion by nkawerau** [oemof/tespy#323] [Q&A]
How does the "init_previous" function work in Tespy? When I set init_previous =True and run a calculation for the second time, I expect to get a solution much faster. But unfortunately this is not the case. For example, the first run to solve a large Network took 367 seconds, the second took 352 seconds. Where are the values stored that init_previous uses? Do you have any other tips to speed up the convergence?

**Comment by fwitte** [comment]
Hi @nkawerau,

I am kind of puzzled by the large network, how big is that? What is the number of components/connections/fluids?

`init_previous` is `True` by default and the starting values from the converged simulation before are taken over. That means, the Newton algorithms is much more likely to find a feasible solution, given the inputs are not too far away from the previous solution. You can make use of this when setting up a model step by step. You'd delete interfacing connections and add the new parts of the network, then rerun the simulation. The untouched part of the network will have the good starting values from the converged solution and therefore convergence is much more stable. That does not necessarily mean, that the overall time is shorter, only that the number of iterations until the calculation converges might be lower. Pseudo-code for the usage of this functionality would look like this

```python
(build some initial network)
some_connection = Connection(...., sink, ...)
(run simulation)
nwk.del_conns(some_connection)
some_connection = Connection(....)
(build more parts of the network)
(run simulation)
```

I'd suggest profiling your script, for example with something like [pyinstrument](https://github.com/joerick/pyinstrument).

Best regards

**Reply by nkawerau** [reply]
Thank you very much for your answer. For sure, "small" or "large network" is subjective. I have to calculate networks with up to 1200 connections, the number of components is similar. I've 4 fluid components, so 7 variables to solve at each connection point.
Is there any difference to `init_previous` if I set the same values for `m0, p0 and h0` or even additionally `T0, v0`, etc...
If you have comments about parallelization, you may read my other post here: https://github.com/oemof/tespy/discussions/324

**Reply by fwitte** [reply]
I read it, but cannot really give an in depth answer. I have never really tried it and am not sure, if the current architecture even supports it.

Well, your Network is huge. Best I've done so far was maybe 1000 variables. Have you tried profiling your code? That should give hints, what can be optimized prior to parallelisation.

> Is there any difference to `init_previous` if I set the same values for `m0, p0 and h0` or even additionally `T0, v0`, etc...

No, actually only `m0`, `p0` and `h0` should have an effect, since `T0`, `v0` etc. are not variables.

If you want, we can have a meeting on Monday and discuss possible improvements.
